packages/valory/skills/oracle_deployment_abci/rounds.py

```diff
@@ -67,9 +67,23 @@ def safe_contract_address(self) -> str:
         return cast(str, self.db.get_strict("safe_contract_address"))
 
     @property
-    def oracle_contract_address(self) -> str:
+    def oracle_contract_address(self) -> Optional[str]:
         """Get the oracle contract address."""
-        return cast(str, self.db.get("oracle_contract_address"))
+        return cast(Optional[str], self.db.get("oracle_contract_address", None))
+
+
+class SetupCheckRound(VotingRound):
+    """A round which checks if the oracle address is already provided via the agents' `setup` or not"""
+
+    round_id = "ffw"
+    allowed_tx_type = VotingOraclePayload.transaction_type
+    payload_attribute = "vote"
+    synchronized_data_class = SynchronizedData
+    done_event = Event.DONE
+    negative_event = Event.NEGATIVE
+    none_event = Event.NONE
+    no_majority_event = Event.NO_MAJORITY
+    collection_key = "participant_to_ffw"
```

Comment:
This round is necessary in order to know if the oracle contract address is provided. Now that the safe contract is always provided via the `setup`, the registration startup always ends with a FFW event, so we need to decide whether we need to deploy the oracle contract or not here.